src/app/src/api/referenceImages.js

```diff
@@ -0,0 +1,80 @@
+import { useDispatch } from 'react-redux';
+import { useTrailingDebounceCallback } from '../hooks';
+import api from './api';
+
+const referenceImagesApi = api.injectEndpoints({
+    endpoints: build => ({
+        uploadReferenceImage: build.mutation({
+            query: ({ boundaryId, ...details }) => ({
+                url: `/boundaries/${boundaryId}/reference-images/`,
+                method: 'POST',
+                data: details,
+            }),
+            onQueryStarted: async (
+                { boundaryId },
+                { dispatch, queryFulfilled }
+            ) => {
+                try {
+                    const { data: newImage } = await queryFulfilled;
+
+                    dispatch(
+                        api.util.updateQueryData(
+                            'getBoundaryDetails',
+                            boundaryId,
+                            draftDetails => {
+                                draftDetails.reference_images.push(newImage);
+                            }
+                        )
+                    );
+                } catch {}
```

Comment:
Should we be handling an error here?

Reply:
It's not clear from [the documentation](https://redux-toolkit.js.org/rtk-query/api/createApi#onquerystarted) if this error will be the same error that is returned from the query function to the mutation hook, but in any case, that hook will still get the error message if there is one.

Here, we don't want to update the cache if the post request fails, so I think it's fine to not do anything.

Reply:
I'll add a comment 👍🏻